### PR TITLE
Thumbproxy

### PR DESCRIPTION
**Pull request by CathyJinJNCC**

### CODE REVIEW

File: www/src/mapper/Thumbnail.tsx [hunk 1]
```diff
@@ -93,23 +101,24 @@ export let Thumb = (props: {
...
+            className="col-span-full row-span-full"
+            style={{height: height, width: width, visibility: loaded? 'visible' : 'hidden'}}
           >
             <img
               src={src}
               className="w-full h-full rounded-md"
               alt={`Thumbnail image for ${getDisplayDate(props.date)}`}
+              onLoad={(e) => {
+                let imageElement = e.target as HTMLElement
+                imageElement.classList.add('animate-quickfadein') // trigger animation when visible
```
Comment (PeteMontgomery) [inline www/src/mapper/Thumbnail.tsx:115]: We shouldn't break out of React if we can help it - shall we have a quick look together at this after merge

**Review by PeteMontgomery — CHANGES_REQUESTED** [review]
Fantastic. Could you add a few explanatory comments where I've noticed, then merge? 👍

File: api/package.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
     "clean": "rimraf ./built",
     "tsc": "tsc --noEmit",
     "build": "tsc --outDir ./built",
-    "dev": "yarn tsc --skipLibCheck && NODE_ENV=development ts-node ./src/host/local.ts",
+    "dev": "yarn tsc --skipLibCheck && NODE_ENV=development ts-node --files ./src/host/local.ts",
```
[the remark this answers is not in the document]
Comment (CathyJinJNCC) [reply api/package.json:9]: I don't know, I think you added that one!

File: api/src/host/aws.ts [hunk 3]
```diff
@@ -56,6 +57,19 @@ export let statsHandler: APIGatewayProxyHandler = async (event) => {
...
+export let thumbHandler: APIGatewayProxyHandler = async (event) => {
+    let result = await getThumb(event.queryStringParameters)
+
+    return {
+      statusCode: 200,
+      headers: {
+        'Content-Type': 'image/png',
+      },
+      body: result.toString('base64'),
+      isBase64Encoded: true
```
[the remark this answers is not in the document]
Comment (CathyJinJNCC) [reply api/src/host/aws.ts:69]: Done

File: api/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (PeteMontgomery) [reply api/package.json:9]: :laughing:

File: api/README.md [hunk 4]
```diff
@@ -20,3 +20,4 @@ Deploy to an AWS environment:
 
     deploy:alpha
 
+Prebuilt lambda layers taken from [node-canvas-lambda](https://github.com/jwerre/node-canvas-lambda).
```
[the remark this answers is not in the document]
Comment (CathyJinJNCC) [reply api/README.md]: Added more explanation

**Review by PeteMontgomery — APPROVED** [review]

**Comment by PeteMontgomery** [comment]
Let's merge 💥

**Pull request merged**